src/main/java/com/cleanroommc/groovyscript/compat/mods/pyrotech/StoneKiln.java

```diff
@@ -0,0 +1,138 @@
+                .failureChance(failureChance)
+                .failureOutput(failureOutput)
+                .name(name)
+                .input(input)
+                .output(output)
+                .register();
+    }
+
+    @MethodDescription
+    public void removeByInput(ItemStack input) {
+        if (GroovyLog.msg("Error removing kiln recipe")
+                .add(IngredientHelper.isEmpty(input), () -> "Input 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneKilnRecipe recipe : getRegistry()) {
+            if (recipe.getInput().test(input)) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @MethodDescription(example = @Example("item('pyrotech:bucket_clay')"))
+    public void removeByOutput(IIngredient output) {
+        if (GroovyLog.msg("Error removing kiln recipe")
+                .add(IngredientHelper.isEmpty(output), () -> "Output 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneKilnRecipe recipe : getRegistry()) {
+            if (output.test(recipe.getOutput())) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "output", valid = @Comp("1"))
+    @Property(property = "name")
+    public static class RecipeBuilder extends AbstractRecipeBuilder<StoneKilnRecipe> {
+
+        @Property
+        private final ItemStackList failureOutput = new ItemStackList();
+        @Property(valid = @Comp(type = Comp.Type.GTE, value = "1"))
+        private int burnTime;
+        @Property(valid = @Comp(type = Comp.Type.GTE, value = "0"))
+        private float failureChance;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder burnTime(int time) {
+            this.burnTime = time;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder failureChance(float chance) {
+            this.failureChance = chance;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder failureOutput(ItemStack failureOutputs) {
+            this.failureOutput.add(failureOutputs);
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder failureOutput(ItemStack... failureOutputs) {
+            for (ItemStack itemStack : failureOutputs) {
+                failureOutput(itemStack);
+            }
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder failureOutput(Iterable<ItemStack> failureOutputs) {
+            for (ItemStack itemStack : failureOutputs) failureOutput(itemStack);
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Pyrotech Kiln Recipe";
```

Comment:
Stone Kiln.
should also correct the corresponding line in the now `PitKiln` file